x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/allocation/CcrPrimaryFollowerAllocationDecider.java

```diff
@@ -57,11 +58,14 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
             return allocation.decision(Decision.YES, NAME,
                 "shard is a primary follower but was bootstrapped already; hence is not under the purview of this decider");
         }
-        if (node.node().isRemoteClusterClient() == false) {
-            return allocation.decision(Decision.NO, NAME, "shard is a primary follower and being bootstrapped, but node does not have the "
-                + DiscoveryNodeRole.REMOTE_CLUSTER_CLIENT_ROLE.roleName() + " role");
+        if (node.node().isRemoteClusterClient()) {
+            return allocation.decision(Decision.YES, NAME,
+                "shard is a primary follower and node has the " + DiscoveryNodeRole.REMOTE_CLUSTER_CLIENT_ROLE.roleName() + " role");
+        }
+        if (node.node().getVersion().before(DiscoveryNode.PLUGGABLE_ROLES_VERSION)) {
```

Comment:
I wonder if this should be the version that we introduced the remote cluster client role in as opposed to the version that we made node roles pluggable.

Reply:
Good catch, fixed in https://github.com/elastic/elasticsearch/pull/60093/commits/3a33f0acd69337e5437dcb911e237582f3a821c6